### PR TITLE
Add tests for element setCustomValidity

### LINKED ISSUE #993: Test for <output>.setCustomValidity()
Proposed test:

http://software.hixie.ch/utilities/js/live-dom-viewer/saved/3026

Checks that <output> can't match :invalid, get an 'invalid' event, or block submission.

### PR DESCRIPTION
**Pull request by xiuqijix**
Add following tests (Fix #993):

- Add tests for output/object/select/textarea setCustomValidity
- Optimize tests for input/button/fieldset setCustomValidity by using element.validity.customError to check if the element has a custom error

@foolip, PTAL.

### CODE REVIEW

File: html/semantics/forms/the-button-element/button-setcustomvalidity.html [hunk 1]
```diff
@@ -1,43 +1,17 @@
...
- <body>
-  <p>
-    <h3>button_setCustomValidity</h3>
-  </p>
+<title>button setCustomValidity</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
 
-  <hr>
+<button id='test'></button>
```
Comment (tkent-google) [inline html/semantics/forms/the-button-element/button-setcustomvalidity.html]: nit: Please avoid the ID 'test'.  Both of this element and testharness's test() can be 'test' property of 'window'.  It's confusing.

File: html/semantics/forms/the-object-element/object-setcustomvalidity.html [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<!DOCTYPE HTML>
```
Comment (tkent-google) [inline html/semantics/forms/the-object-element/object-setcustomvalidity.html]: We should not create the-object-element/ directory under semantics/forms/.  It should be in semantics/embedded-content/.  The directory structure in html/ should be matched to the structure of the HTML standard.

**Review by tkent-google — CHANGES_REQUESTED** [review]

File: html/semantics/forms/the-button-element/button-setcustomvalidity.html (hunk 1 above)
Comment (kaixinjxq) [reply html/semantics/forms/the-button-element/button-setcustomvalidity.html]: @tkent-google, many thanks for your comments, I've addressed it, PTAL.

File: html/semantics/forms/the-object-element/object-setcustomvalidity.html (hunk 2 above)
Comment (kaixinjxq) [reply html/semantics/forms/the-object-element/object-setcustomvalidity.html]: Updated

**Review by tkent-google — APPROVED** [review]

**Pull request merged**